community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java

```diff
@@ -820,6 +820,18 @@
     public static final Setting<Duration> bookmark_ready_timeout = buildSetting(
             "dbms.transaction.bookmark_ready_timeout", DURATION, "30s" ).constraint( min( Duration.ofSeconds( 1 ) ) ).build();
 
+    public enum TransactionStateMemoryAllocation
+    {
+        ON_HEAP,
+        OFF_HEAP
+    }
+
+    @Description( "Defines where to allocate memory for storing transaction state. Valid values are `on_heap` and `off_heap`." )
+    public static final Setting<TransactionStateMemoryAllocation> tx_state_memory_allocation = buildSetting(
```

Comment:
lets make this property internal and lets change its namespace to unsupported for now since general public should be use it yet. Also description should be updated to highlight that its experimental feature

Reply:
✅